File: src/main/java/loxal/epvin/datastore/service/DAOServiceLocator.java
/*
 * Copyright 2012 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package loxal.epvin.datastore.service;

import com.google.web.bindery.requestfactory.shared.ServiceLocator;

/**
 * Must be in the same package as the DAOService classes.
 */
public class DAOServiceLocator implements ServiceLocator {
    @Override
    public Object getInstance(final Class<?> cls) {
        try {
            return cls.newInstance();
        } catch (final InstantiationException e) {
            throw new RuntimeException(e);
        } catch (final IllegalAccessException e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/kotlin/loxal/epvin/datastore/service/DAOServiceLocator.kt
/*
 * Copyright 2012 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package loxal.epvin.datastore.service

import com.google.web.bindery.requestfactory.shared.ServiceLocator

/**
 * Must be in the same package as the DAOService classes.
 */
public class DAOServiceLocator : ServiceLocator {
    override fun getInstance(cls: Class<*>): Any? {
        try {
            return cls.newInstance()
        } catch (e: InstantiationException) {
            throw RuntimeException(e)
        } catch (e: IllegalAccessException) {
            throw RuntimeException(e)
        }

    }
}